macros/get_metric_sql.sql

```diff
@@ -57,24 +97,32 @@ LETS SET SOME VARIABLES AND VALIDATE!
 a custom calendar #}
 {%- set calendar_tbl = ref(var('dbt_metrics_calendar_model', "dbt_metrics_default_calendar")) %}
 
-{# Here we are creating a list of all valid dimensions, as well as providing compilation
-errors if there are any provided dimensions that don't work. #}
-{% set common_valid_dimension_list = metrics.get_common_valid_dimension_list(dimensions, metric_tree['full_set']) %}
+{% if is_develop_macro %}
+    {%- set calendar_dimensions = metrics.get_calendar_dimension_list(metric_dimensions, dimensions) -%}
+    {%- set non_calendar_dimensions = metrics.get_non_calendar_dimension_list(dimensions) -%}
+    {%- set relevant_periods = metrics.get_relevent_periods(grain, secondary_calculations) %}
 
-{# We have to break out calendar dimensions as their own list of acceptable dimensions. 
-This is because of the date-spining. If we don't do this, it creates impossible combinations
-of calendar dimension + base dimensions #}
-{%- set calendar_dimensions = metrics.get_calendar_dimension_list(dimensions, common_valid_dimension_list) -%}
+{% else %}
+    {# Here we are creating a list of all valid dimensions, as well as providing compilation
+    errors if there are any provided dimensions that don't work. #}
+    {% set common_valid_dimension_list = metrics.get_common_valid_dimension_list(dimensions, metric_tree['full_set']) %}
+
+    {# We have to break out calendar dimensions as their own list of acceptable dimensions. 
+    This is because of the date-spining. If we don't do this, it creates impossible combinations
+    of calendar dimension + base dimensions #}
+    {%- set calendar_dimensions = metrics.get_calendar_dimension_list(dimensions, common_valid_dimension_list) -%}
```

Comment:
`calendar_dimensions` here depends on `common_valid_dimension_list` or just `dimensions`. You'll never guess my question! 

Does this _have_ to be split out? What would happen if you asked the macros to `get_common_valid_dimension_list` for the develop set, assuming you pass in a sufficiently-developed metric tree?

Reply:
This section was in need of a MAJOR rework. You're totally right in that it doesn't really make sense - `get_common_valid_dimension_list` was just validating and not creating any variables. I've updated this as part of the work done in #86 because it was relevant to the work being done there around dimension validation and table introspection